ios/browser/brave_ads/ads_service_impl_ios.mm

```diff
@@ -75,6 +77,70 @@
                                 std::move(callback)));
 }
 
+void AdsServiceImplIOS::RunDBTransaction(
+    mojom::DBTransactionInfoPtr mojom_db_transaction,
+    RunDBTransactionCallback callback) {
+  database_.AsyncCall(&brave_ads::Database::RunDBTransaction)
+      .WithArgs(std::move(mojom_db_transaction))
+      .Then(std::move(callback));
+}
+
+void AdsServiceImplIOS::MaybeGetNotificationAd(
+    const std::string& placement_id,
+    MaybeGetNotificationAdCallback callback) {
+  if (!IsRunning()) {
+    return std::move(callback).Run(/*ad*/ std::nullopt);
+  }
+
+  ads_->MaybeGetNotificationAd(placement_id, std::move(callback));
+}
+
+void AdsServiceImplIOS::TriggerNotificationAdEvent(
+    const std::string& placement_id,
+    mojom::NotificationAdEventType mojom_ad_event_type,
+    TriggerAdEventCallback callback) {
+  if (!IsRunning()) {
+    return std::move(callback).Run(/*success*/ false);
+  }
+
+  ads_->TriggerNotificationAdEvent(placement_id, mojom_ad_event_type,
+                                   std::move(callback));
+}
+
+void AdsServiceImplIOS::AddObserver(AdsServiceObserver*
+                                    /*observer*/) {
+  NOTIMPLEMENTED();
+}
+
+void AdsServiceImplIOS::RemoveObserver(AdsServiceObserver*
+                                       /*observer*/) {
+  NOTIMPLEMENTED();
+}
+
+bool AdsServiceImplIOS::IsBrowserUpgradeRequiredToServeAds() const {
+  NOTIMPLEMENTED();
+  return {};
+}
+
+int64_t AdsServiceImplIOS::GetMaximumNotificationAdsPerHour() const {
+  NOTIMPLEMENTED();
+  return {};
+}
+
+void AdsServiceImplIOS::OnNotificationAdShown(const std::string& placement_id) {
+  NOTIMPLEMENTED();
+}
+
+void AdsServiceImplIOS::OnNotificationAdClosed(const std::string& placement_id,
+                                               const bool by_user) {
+  NOTIMPLEMENTED();
+}
+
+void AdsServiceImplIOS::OnNotificationAdClicked(
+    const std::string& placement_id) {
```

Comment:
nit: comment out params that are not used, i.e., `/*placement_id*/`. Ditto for other functions too.